gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -1910,7 +1910,15 @@ accl_hw_get_type (void)
       {ACCL_DEFAULT, ACCL_DEFAULT_STR, ACCL_DEFAULT_STR},
       {ACCL_AUTO, ACCL_AUTO_STR, ACCL_AUTO_STR},
       {ACCL_CPU, ACCL_CPU_STR, ACCL_CPU_STR},
+#if defined(__aarch64__) || defined(__arm__)
+      /** Retreive NEON_STR when searching for SIMD/NEON on arm architectures */
       {ACCL_CPU_NEON, ACCL_CPU_NEON_STR, ACCL_CPU_NEON_STR},
+      {ACCL_CPU_SIMD, ACCL_CPU_SIMD_STR, ACCL_CPU_SIMD_STR},
+#else
+      /** Retreive SIMD_STR when searching for NEON/SIMD on x86 architectures */
+      {ACCL_CPU_SIMD, ACCL_CPU_SIMD_STR, ACCL_CPU_SIMD_STR},
+      {ACCL_CPU_NEON, ACCL_CPU_NEON_STR, ACCL_CPU_NEON_STR},
```

Comment:
If it's not ARM, let's not convert 0x1100 to NEON.

Reply:
Did you mean - NEON entry should not be there if it is not ARM?
then the user requesting NEON in non-ARM will not be forwarded to SIMD.

Reply:
It's ok to emit errors or ignore if neon is designated for non-ARM machines. They do not have NEON.

SIMD->NEON "translation" is fine, but NEON->SIMD "translation" in non-ARM may be dangerous. Assume that there is a software code that supports NEON but no SSE and will call NEON instructions unconditionally if "neon" is designated (unsafe but not incorrect).

Reply:
if-ARM section looks good: it states that "neon" "simd" will be translated as NEON and SIMD.
if-x64 section might be undesirable: it will support "neon" in x64. (although it's just X64-SIMD actually.).

Reply:
That is a very good example. NEON->SIMD is wrong. I will remove this translation, and update x64 section to just have SIMD (neon will not match with SIMD).
This will also impose restrictions on subplugins to not add NEON in supported accelerators list in non-arm architectures (removed ACCL_CPU_NEON_STR in non-arm architectures to check on this).